### ISSUE TITLE
Javascript - Win or Tie Game Conditions 

### ISSUE THREAD

**Issue by RoobScoobs** [ga-wdi-boston/game-project#197]
Happy Saturday guys, 

Wondering if the game object, which is now receiving the patch update, is somehow accessible to construct the win or tie conditions.

My line of thinking is to extract the values at each cell and check them for rows, columns and diagonals. 

``` js
// edited by @jrhorn424
const winOrTieGame = function () {
    if (game.cells[0] && game.cells[1] && game.cells[2] === player1) {
      console.log("X is the Winner!"); 
```

... followed by a bunch of ELSE IF statements that check the other 7 winning combinations

**Comment by RoobScoobs** [comment]
Whoa... I just did something, and it ALMOST returned what I wanted. I managed to push the value (X or O) into the array, however, the value is always added to the first index in the array. 

``` js
// edited by @jrhorn424
gameArray.insert = function (index, move) {
  this.splice(index, 0, move);
};
```

This function allows me to position an element anywhere I want in the array 

The function below allows me to pull out the string in my html id and convert it to an integer, followed by changing it to its appropriate index value.  

``` js
const makeAMove = function (event) {
    event.preventDefault();
    let position = event.target.id;
    let index = function () {
      return parseInt(position.slice(-1)) - 1;
    };
    if ($(this).text() !== ''){
      window.alert('Make a valid move');
    } else if (turnCount % 2 === 0) {
        $(this).text(player1);
        move = player1;
        turnCount++;
        // winOrTieGame();
    } else {
        $(this).text(player2);
        move = player2;
        turnCount++;
        // winOrTieGame();
    }
    gameArray.insert(index, move);
    console.log(gameArray);
};
```

**Comment by berziiii** [comment]
Did you figure it out, @RoobScoobs?

**Comment by RoobScoobs** [comment]
Yes, totally did just now

**Comment by gaand** [comment]
@RoobScoobs Please explain your solution when closing an issue.  It can help your peers when they encounter similar problems, and it can help you remember it better if you write it down in words and not just code.

**Comment by jrhorn424** [comment]
@RoobScoobs What was the resolution?

These notes are for future developers referencing this issue:

Use fenced code blocks to paste code. Fenced code blocks begin with three backticks (or grave accents – &nbsp;&grave;&nbsp;) and are followed by the filetype. In this case, you probably want `js`. So you'd begin a code block with (&nbsp;&grave;&grave;&grave;js&nbsp;). Fenced code blocks are closed with three more backticks, with no filetype specifier (  &nbsp;&grave;&grave;&grave;&nbsp; ).

Use a single backtick to do an inline code snippet. It must begin and end with one backtick, and you cannot specify a filetype. `like this`

CC for information only @ga-wdi-boston/core @RoobScoobs

**Comment by RoobScoobs** [comment]
The code noted from above never worked because I was attaching a function to an array rather than defining a function and passing it an array. 

Instead of writing this: 

``` js

gameArray.insert = function (index, move) {
  this.splice(index, 1, move);
};

```

The function should be written as this: 

``` js

let insert = function (array, index, move) {
  array.splice(index, 1, move);
};

```

Then when it's called I can pass it the empty array defined as  `let gameArray = ['','','','','','','','']`

Changing `array.splice(index, 0 , move)` to `array.splice(index, 1, move)` will prevent additional items from being added to the array. 

As for fixing the index value, I decided to scratch the function attached to the index and rewrite the html div containers to reflect a unique id (starting from 0 - 8). Then, I redefined my index to: 

``` js

let index = $(event.target).attr('id');

```
